messages/main.go

```diff
@@ -22,51 +24,82 @@ type Response events.APIGatewayProxyResponse
 // Request is of type APIGatewayProxyRequest
 type Request events.APIGatewayProxyRequest
 
+var slackBotUserOauthToken = os.Getenv("SLACK_BOT_USER_OAUTH_ACCESS_TOKEN")
 var slackOauthToken = os.Getenv("SLACK_OAUTH_ACCESS_TOKEN")
-var slackSigningSecret = os.Getenv("SLACK_SIGNING_SECRET")
-var api = slack.New(slackOauthToken)
+var slackSigningSecret = os.Getenv("SLACK_VERIFICATION_TOKEN")
+var botApi = slack.New(slackBotUserOauthToken)
+
+// var regularApi = slack.New(slackOauthToken)
 
 // Handler is our lambda handler invoked by the `lambda.Start` function call
 func Handler(request Request) (Response, error) {
-	buf := new(bytes.Buffer)
 	body := []byte(request.Body)
+	log.Printf("Receiving request body %s", body)
 	resp := Response{
 		IsBase64Encoded: false,
 		Headers: map[string]string{
 			"Content-Type": "application/json",
 		},
 	}
-	eventsAPIEvent, e := slackevents.ParseEvent(json.RawMessage(body),
-		slackevents.OptionVerifyToken(&slackevents.TokenComparator{VerificationToken: slackSigningSecret}))
+	eventsAPIEvent, e := slackevents.ParseEvent(
+		json.RawMessage(body),
+		slackevents.OptionVerifyToken(&slackevents.TokenComparator{VerificationToken: slackSigningSecret}),
+	)
 	if e != nil {
+		log.Println("Could not parse Slack event :'(")
 		resp.StatusCode = 500
 	}
+	log.Printf("Processing an event of outer type %s", eventsAPIEvent.Type)
 
-	if eventsAPIEvent.Type == slackevents.Message {
-		var r *slackevents.ChallengeResponse
-		err := json.Unmarshal(body, &r)
-		if err != nil {
-			resp.StatusCode = 500
-		}
-		resp.Headers["Content-Type"] = "text"
-		buf.Write([]byte(r.Challenge))
-		resp.Body = buf.String()
-	}
-	if eventsAPIEvent.Type == slackevents.CallbackEvent {
+	handleSlackChallenge(eventsAPIEvent, body, resp)
+
+	if eventsAPIEvent.Type == slackevents.CallbackEvent || eventsAPIEvent.Type == slackevents.AppMention {
 		innerEvent := eventsAPIEvent.InnerEvent
+		log.Printf("Processing an event of inner data %s", innerEvent.Data)
 		switch ev := innerEvent.Data.(type) {
 		case *slackevents.AppMentionEvent:
-			api.PostMessage(ev.Channel, slack.MsgOptionText("Yes, hello.", false))
+			resp.StatusCode = 200
+			fmt.Printf("Reacting to app mention event from channel %s", ev.Channel)
```

Comment:
any reasons why we use both `log.Printf` and `fmt.Printf` in this file?

Reply:
Good point. I think using log is a better practice, as I think it'll be easier to format the log later.